corehq/util/files.py

```diff
@@ -40,3 +42,25 @@ def safe_filename_header(filename, extension=None):
     ascii_filename = unidecode(filename)
     return 'attachment; filename="{}"; filename*=UTF-8\'\'{}'.format(
         ascii_filename, quote(filename.encode('utf8')))
+
+
+class TransientTempfile(object):
+    """
+    Manage a temporary file that can be opened and closed as needed, but will
+    be automatically cleaned up on failure or exit.
+
+        with TransientTempfile() as path:
+            # path exists throughout this block
+            with open(path, 'w') as f:
+                f.write("Adding stuff to the file")
+            with open(path) as f:
+                do_stuff(f.read())
+        # path no longer exists
+    """
+
+    def __enter__(self):
+        _, self.path = tempfile.mkstemp()
```

Comment:
I think you want to call `.close()` on thing you're throwing away here (`_`).

Reply:
Oh yikes, you're right.  We do this all over the place.  Updated.